code/modules/reagents/chemistry/machinery/chem_recipe_debug.dm

```diff
@@ -0,0 +1,103 @@
+/*
+* A debug chem tester that will process through all recipies automatically and try to react them.
+* Highlights low purity reactions and and reactions that don't happen
+*/
+/obj/machinery/chem_recipe_debug
+	name = "chemical reaction tester"
+	density = TRUE
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "HPLC"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 40
+	resistance_flags = FIRE_PROOF | ACID_PROOF | INDESTRUCTIBLE
+	var/list/cached_reactions = list()
+	var/index = 1
+	var/processing = FALSE
+	var/problem_string
+	var/minorImpurity
+	var/majorImpurity
```

Comment:
If you could give brief descriptions on what each variable is about, that'd be positive.

Reply:
I overlooked it because it being a debug machine. I've rectified this now.